plugins/woocommerce-admin/client/inbox-panel/index.js

```diff
@@ -238,15 +299,17 @@ const InboxPanel = ( { showHeader = true } ) => {
 
 		const noteId = note.id;
 		try {
-			removeNote( noteId );
+			await removeNote( noteId );
+			invalidateResolutionForStoreSelector( 'getNotes' );
```

Comment:
It seems we need to invalidate the data selector manually; I'm assuming this is due to the the fact that we're changing the query, but I haven't dived into too deeply. Let me know if you see a more elegant way to manage this.

Reply:
Yeah that makes sense, I don't think there is a much better way around this at the moment, unless we add a new selector or change the selector to either ignore `per_page` prop when returning the items, or use a separate selector that just returns all the `notes`.
But this is fine!

Reply:
I think this will result in the simplest way to give the "rolling" notes appearance as we dismiss/etc notes. I'll keep it as is for the moment.